tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,524 @@
+    test.same(rootEditionsOverriden.lookup("Message").fields.stringRepeated._features, {
+        enum_type: 'CLOSED',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' }
+    })
+    
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to nested message", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023"; 
+    message Message {
+        option features.json_format = LEGACY_BEST_EFFORT;
+        option features.(abc).d_e = deeply_nested_false;
+        string string_val = 1;
+        string string_repeated = 2 [features.enum_type = CLOSED];
+
+    message Nested {
+        option features.(abc).d_e = deeply_nested_true;
+        option features.field_presence = IMPLICIT;
+        int64 count = 9;
+    }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup("Message").lookup("Nested")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'IMPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_true' } 
+    });
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to enums and enum values", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+
+    option features.(abc).d_e = deeply_nested_false;
+    message Message {
+        enum SomeEnum {
+            ONE = 1 [features.repeated_field_encoding = EXPANDED];
+            TWO = 2;
+        }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookupEnum("SomeEnum")._valuesFeatures["ONE"], {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'EXPANDED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.same(rootEditionsOverriden.lookupEnum("SomeEnum")._valuesFeatures["TWO"], {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    });
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to oneofs", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`
+    edition = "2023";
+    option features.(abc).d_e = deeply_nested_false;
+    message Message {
+        option features.json_format = LEGACY_BEST_EFFORT;
+        oneof SomeOneOf {
+            int32 a = 13;
```

Comment:
you can put features in oneofs (same syntax as for message), do those work?

Reply:
Following the pattern of your other tests, I'd expect these two tests to be collapsed into one that tests both inheritance and overriding of message -> oneof

Reply:
isn't that tested in the test below? (`feature resolution inheritance oneofs`)

Reply:
Kindof, it's not clear why this one is split into two when all of the other ones test both things in a single test

Reply:
ok I merged it into one test!